Dependency updates (needs compatibility validation

CyberFlameGO opened the pull request.

Elierrr commented on NachoSpigot-API/pom.xml:
```diff
@@ -10,29 +10,19 @@
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>15</maven.compiler.source>
+        <maven.compiler.target>15</maven.compiler.target>
```
https://github.com/CobbleSword/NachoSpigot/pull/159#discussion_r707199048

Elierrr replied:
You only changed the server back to Java 8 and now the server won't compile :p

CyberFlameGO replied:
Gadzooks! I'll fix that now haha, I must've missed it

LewUwU replied:
Instead of dropping java 8 without any reason, do a issue about why nachospigot should only support java 15-17

Elierrr replied:
It was on accident and already corrected.

Elierrr commented on NachoSpigot-API/pom.xml:
```diff
@@ -10,29 +10,19 @@
 
     <artifactId>api</artifactId>
-    <version>1.8.8-R0.2-SNAPSHOT</version>
+    <version>1.8.8-R0.3-SNAPSHOT</version>
```
Why the change?

CyberFlameGO replied:
That's actually the latest 1.8.8 release, I'm pretty sure putting it as `R0.2` was an accident

Elierrr replied:
Spigot and Paper both have RO.1 but TacoSpigot [changed it](https://github.com/TacoSpigot/TacoSpigot/commit/816cde55dfc03d813ca55e0d66a49eb1fae11f21#diff-3e189d8b27d4e715e93f4f3e6f2875558d4f3f01d5a4526c9b7a1eed3dff3d54) to RO.2. I have no idea how the versioning works but I think it should be RO.1

LewUwU replied:
The sub-release of a version

Elierrr replied:
Then it should be RO.1

Elierrr reviewed (changes requested):
Everything else seems fine.

CyberFlameGO commented:
https://github.com/CobbleSword/NachoSpigot/pull/160#discussion_r707219371 fixed this just before you sent it :)

Again, for the force-pushes, they were to amend commit messages (but this time not signing)

CyberFlameGO commented:
Another force-push which was necessary to amend a commit, and another regular commit was added as per https://github.com/CobbleSword/NachoSpigot/pull/160#discussion_r707234069

Elierrr commented:
LGTM, I'll test it when I can.

andreasdc commented:
What is the difference between R0.3 and R0.2?

CyberFlameGO commented:
I could've sworn I've seen a R0.3 before; apparently not then. My bad

On Tue, 14 Sep 2021 at 12:38 AM, Elier ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In NachoSpigot-API/pom.xml
> <https://github.com/CobbleSword/NachoSpigot/pull/160#discussion_r707292811>
> :
>
> > @@ -10,29 +10,19 @@
>      </parent>
>
>      <artifactId>api</artifactId>
> -    <version>1.8.8-R0.2-SNAPSHOT</version>
> +    <version>1.8.8-R0.3-SNAPSHOT</version>
>
> Spigot and Paper both have RO.1 but TacoSpigot changed it
> <https://github.com/TacoSpigot/TacoSpigot/commit/816cde55dfc03d813ca55e0d66a49eb1fae11f21#diff-3e189d8b27d4e715e93f4f3e6f2875558d4f3f01d5a4526c9b7a1eed3dff3d54>
> to RO.2. I have no idea how the versioning works but I think it should be
> RO.1
>
> —
> You are receiving this because you authored the thread.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/CobbleSword/NachoSpigot/pull/160#discussion_r707292811>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF6BVMFV6XWNKEMKQNWJ6G3UBXWCPANCNFSM5D5QU4HA>
> .
>

Lucaskyy commented:
I've spoken with @LewUwU before about this, this sadly breaks a lot of plugins. SnakeYAML will completely stop working for nearly every plugin on the latest version, due to strict security measures in the new version which is a pain to fix. And the MySQL Connector also shouldn't be updated since that also breaks a lot of plugins. As long as the dependency is internal you can update it without any issues, but as for external dependencies please watch out. And indeed, in my opinion, it's better to leave it R0.2 for compatibility reasons and in case something stupid happens.

andreasdc commented:
> n

What's the difference betwen r3 and r2?

LewUwU reviewed (changes requested):
R0.3 -> R0.2, you're breaking tons of things

andreasdc commented:
> R0.3 -> R0.2, you're breaking tons of things

like?

Lucaskyy commented:
NMS plugins mostly.

andreasdc commented:
But package is currently v1_8_R3

Lucaskyy commented:
I understand, but a lot of plugins look at the version string v1_8_R3 and determine that way that a server is running 1.8. Changing that breaks that. Don't do that.

andreasdc commented:
> I understand, but a lot of plugins look at the version string v1_8_R3 and determine that way that a server is running 1.8. Changing that breaks that. Don't do that.

You could always recode these plugins

Lucaskyy commented:
What the heck? No? Me changing a tiny number requires me to recode tons of plugins? No thanks.

andreasdc commented:
> What the heck? No? Me changing a tiny number requires me to recode tons of plugins? No thanks.

People that want R4 would have to recode, that's not big problem. I personally use only plugins coded by myself.
What changing from r3 to r4 brings?

Lucaskyy commented:
It has nothing to do with compiling.
Some plugins read the package name (net.minecraft.v1_8_R3) and changing it will break a lot. Not happening. End of discussion.

andreasdc commented:
> .

Yea but does it bring any improvement?

Lucaskyy commented:
R3 to R4? No, absolutely not lol. All it does it break things

andreasdc commented:
All right, I get it. Are changes viewable somewhere maybe?

Lucaskyy commented:
Which changes?

andreasdc commented:
From R3 to R4

Lucaskyy commented:
Well the version was never supposed to be bumped anyway or whatever happened to it. So, just ignore it.

CyberFlameGO commented:
I don't actually think I touched what you're referring to. I changed some
XML tag at the top, not the NMS package

On Fri, 17 Sep 2021 at 10:11 AM, Sculas ***@***.***> wrote:

> Well the version was never supposed to be bumped anyway or whatever
> happened to it. So, just ignore it.
>
> —
> You are receiving this because you authored the thread.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/CobbleSword/NachoSpigot/pull/160#issuecomment-921289458>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF6BVMGPW2V2FRCHZUOGQBTUCJTQDANCNFSM5D5QU4HA>
> .
>

CyberFlameGO commented:
If that's what you mean, you can fix it yourself because I've given
maintainers permission to act on the PR

On Fri, 17 Sep 2021 at 11:24 AM, Aaron Lowe ***@***.***> wrote:

> I don't actually think I touched what you're referring to. I changed some
> XML tag at the top, not the NMS package
>
> On Fri, 17 Sep 2021 at 10:11 AM, Sculas ***@***.***> wrote:
>
>> Well the version was never supposed to be bumped anyway or whatever
>> happened to it. So, just ignore it.
>>
>> —
>> You are receiving this because you authored the thread.
>>
>>
>> Reply to this email directly, view it on GitHub
>> <https://github.com/CobbleSword/NachoSpigot/pull/160#issuecomment-921289458>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AF6BVMGPW2V2FRCHZUOGQBTUCJTQDANCNFSM5D5QU4HA>
>> .
>>
>

CyberFlameGO commented:
> I've spoken with @LewUwU before about this, this sadly breaks a lot of plugins. SnakeYAML will completely stop working for nearly every plugin on the latest version, due to strict security measures in the new version which is a pain to fix. And the MySQL Connector also shouldn't be updated since that also breaks a lot of plugins. As long as the dependency is internal you can update it without any issues, but as for external dependencies please watch out. And indeed, in my opinion, it's better to leave it R0.2 for compatibility reasons and in case something stupid happens.

Fixed although I changed us to R0.1 in accordance with Spigot/Paper upstream, as apparently TacoSpigot changed it

CyberFlameGO commented:
https://github.com/CobbleSword/NachoSpigot/pull/160#discussion_r707560120 will do :)

Elierrr commented on NachoSpigot-Server/pom.xml:
```diff
@@ -60,12 +58,12 @@
             <groupId>it.unimi.dsi</groupId>
             <artifactId>fastutil</artifactId>
-            <version>8.5.4</version>
+            <version>8.5.5</version>
```
```suggestion
            <version>8.5.6</version>
```

Elierrr replied:
```suggestion
            <version>8.5.6</version>
```

Elierrr commented on NachoSpigot-Server/pom.xml:
```diff
@@ -245,7 +243,7 @@
                                 <relocation>
                                     <pattern>net.minecraft.server</pattern>
-                                    <shadedPattern>net.minecraft.server.v${minecraft_version}</shadedPattern>
+                                    <shadedPattern>net.minecraft.server.v1_8_R3</shadedPattern>
```
On line 231
```patch
-                                    <shadedPattern>org.bukkit.craftbukkit.v${minecraft_version}</shadedPattern>
+                                    <shadedPattern>org.bukkit.craftbukkit.v1_8_R3</shadedPattern>
```

CyberFlameGO replied:
what's this? i just removed the variable at the top and put its content where it was used, which is there

Elierrr replied:
I know, but you forgot to change it for craftbukkit's relocation.
